.gitmodules

```diff
@@ -1,4 +1,4 @@
 [submodule "llvm"]
 	path = llvm
-	url = ../../circt/llvm.git
+	url = https://github.com/llvm/llvm-project.git
```

Comment:
Does it work to do `../../../llvm/llvm-project.git` in order to inherit whether the original repo was clone via https or ssh?

Reply:
I *think* this will work, both for local and on GitHub workflows. I'm not sure where I got this hunch, but I think on GitHub, the workflow is smart enough to know that `..` means "go up one 'directory' to the llvm org, and down into the llvm-project repo". For local, as long as you checkout llvm-project alongside circt, it should work.

Reply:
Wouldn't that mean anyone forking circt would also have to fork llvm (or this stuff locally)?

Reply:
If you clone a fork, then maybe?

I think the better approach would be to suggest users to set local settings if they want to deviate from this.  It looks like there is a git config `url.<base>.insteadOf` that can be used to, e.g., rewrite `https` urls to some other protocol. 

`https` is a sensible default. I'm good with this as is.

Reply:
insteadOf is really the best way to solve this issue that I've found.

Reply:
Also, what if a user wants to fork to a non-github service? Relative paths won't work. I think we've had that objection in the past in addition to the https vs ssh issue. I haven't found a way to specify a full URL without the protocol specifier to let git select the protocol.

@seldridge RE inheriting https vs ssh: although I typically use SSH, it's my understanding that the new "smart" https behavior is to use https to do reads but do some fancy https/ssh negotiation on pushes.